src/libs/tmdk/tests/deploy.test.ts

```diff
@@ -1,26 +1,265 @@
+    twinmakerMock
+      .on(GetWorkspaceCommand)
+      .rejects(new ResourceNotFoundException({ $metadata: {}, message: "" }));
+
+    const argv2 = {
+      _: ["init"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": "non-existent",
+      dir: fakeTmdkDir,
+    } as Arguments<Options>;
+    await expect(handler(argv2)).rejects.toThrow(ResourceNotFoundException);
+  });
+
+  test("deploy_givenEmptyProject_expectSuccess", async () => {
+    fs.existsSync = jest.fn(() => {
+      return true;
+    });
+    fs.readFileSync = jest.fn((path: [string]) => {
+      if (path.includes("tmdk")) {
+        return JSON.stringify(emptyTmdk, null, 4);
+      } else {
+        // covers entities.json
+        return JSON.stringify([], null, 4);
+      }
+    });
+    twinmakerMock.on(GetWorkspaceCommand).resolves({});
+
+    const argv2 = {
+      _: ["init"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": workspaceId,
+      dir: fakeTmdkDir,
+    } as Arguments<Options>;
+    expect(await handler(argv2)).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateComponentTypeCommand).length).toBe(
+      0
+    );
+    expect(twinmakerMock.commandCalls(CreateEntityCommand).length).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateSceneCommand).length).toBe(0);
+  });
+
+  test("deploy_given1ComponentType_expectSuccess", async () => {
+    fs.existsSync = jest.fn(() => {
+      return true;
+    });
+    fs.readFileSync = jest.fn((path: [string]) => {
+      if (path.includes("tmdk")) {
+        const tmdk1Ct = JSON.parse(JSON.stringify(emptyTmdk));
+        tmdk1Ct["component_types"] = ["componentType1.json"];
+        return JSON.stringify(tmdk1Ct, null, 4);
+      } else if (path.includes("componentType1.json")) {
+        return JSON.stringify(componentType1Definition, null, 4);
+      } else {
+        // covers entities.json
+        return JSON.stringify([], null, 4);
+      }
+    });
+    twinmakerMock.on(GetWorkspaceCommand).resolves({});
+    twinmakerMock
+      .on(GetComponentTypeCommand)
+      .rejectsOnce(
+        new ResourceNotFoundException({ $metadata: {}, message: "" })
+      )
+      .resolves({});
+
+    const argv2 = {
+      _: ["init"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": workspaceId,
+      dir: fakeTmdkDir,
+    } as Arguments<Options>;
+    expect(await handler(argv2)).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateComponentTypeCommand).length).toBe(
+      1
+    );
+    expect(
+      twinmakerMock.commandCalls(CreateComponentTypeCommand)[0].args[0].input
+    ).toStrictEqual({
+      workspaceId: workspaceId,
+      ...componentType1Definition,
+    });
+    expect(twinmakerMock.commandCalls(CreateEntityCommand).length).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateSceneCommand).length).toBe(0);
+  });
+
+  test("deploy_given1SceneAnd1Model_expectSuccess", async () => {
```

Comment:
better test name: `it('deploys successfully when there is one scene and one model')`

Reply:
Got it, will update